inference-engine/src/mkldnn_plugin/mkldnn_node.h

```diff
@@ -669,7 +674,30 @@ class MKLDNNNode {
 
     bool canBePerformedAsScaleShift(const MKLDNNNode *parentNode = nullptr) const;
 
+    bool isDynamicNode() const {
+        return isDynamic;
+    }
+
+    Shape getOriginalInputShapeAtPort(size_t port) const {
+        if (inputShapes.size() <= port) {
+            IE_THROW() << "Incorrect input port number for node " << getName();
+        }
+        return inputShapes[port];
+    }
+
+    Shape getOriginalOutputShapeAtPort(size_t port) const {
+        if (outputShapes.size() <= port) {
+            IE_THROW() << "Incorrect output port number for node " << getName();
+        }
+        return outputShapes[port];
+    }
+
 protected:
+    virtual std::vector<std::vector<size_t>> shapeInfer() const {
```

Comment:
What do you think, is not it better to leave `MKLDNNNode::redefineOutputMemory(std::vector<std::vector<size_t>> newShapes)` only for manual setting of the output shape (m.b. only for the `Input` node), and combine the `shapeInfer` call with the output memory desc redefinition for all other node types?

Reply:
done